modulemd/tests/test-modulemd-merger.c

```diff
@@ -102,6 +104,164 @@ merger_test_deduplicate (CommonMmdTestFixture *fixture,
+  g_assert_true (g_strv_contains (httpd_profile_streams,"notreal"));
+  g_clear_pointer(&httpd_profile_streams,g_strfreev);
+
+  g_assert_cmpstr(modulemd_defaults_v1_get_default_stream(httpd_defaults,"workstation"), == ,"2.4");
+  
+  httpd_profile_streams = modulemd_defaults_v1_get_streams_with_default_profiles_as_strv(httpd_defaults,"workstation");
+  g_assert_cmpint (g_strv_length (httpd_profile_streams), ==, 2);
+  g_assert_true(g_strv_contains(httpd_profile_streams,"2.4"));
+  g_assert_true(g_strv_contains(httpd_profile_streams,"2.6"));
+  g_clear_pointer(&httpd_profile_streams,g_strfreev);
+  httpd_profile_streams = modulemd_defaults_v1_get_default_profiles_for_stream_as_strv (httpd_defaults,"2.4", "workstation");
+  g_assert_cmpint (g_strv_length (httpd_profile_streams), ==, 1);
+  g_clear_pointer(&httpd_profile_streams,g_strfreev);
+  httpd_profile_streams = modulemd_defaults_v1_get_default_profiles_for_stream_as_strv (httpd_defaults,"2.6", "workstation");
+  g_assert_cmpint (g_strv_length (httpd_profile_streams), ==, 3);
+  g_clear_pointer(&httpd_profile_streams,g_strfreev);
+
+  /*
+   * Get another set of objects that will override the default stream for
+   * nodejs 
+   */
+  
+  yaml_path = g_strdup_printf ("%s/overriding-nodejs.yaml", g_getenv ("TEST_DATA_PATH"));
+  override_nodejs_index = modulemd_module_index_new ();
+  modulemd_module_index_update_from_file (override_nodejs_index, yaml_path, TRUE, &failures,&error);
+  g_clear_pointer(&yaml_path,g_free);
+
+  /*
+   * Test that adding both of these at the same priority level results in
+   * the no default stream
+   */
+  merger = modulemd_module_index_merger_new ();
+  modulemd_module_index_merger_associate_index(merger,base_index, 0);
+  modulemd_module_index_merger_associate_index(merger,override_nodejs_index, 0);
+  
+  merged_index = modulemd_module_index_merger_resolve(merger,NULL);
+  g_clear_object(&merger);
+
+  nodejs = modulemd_module_index_get_module(merged_index,"nodejs");
+  g_assert_nonnull(nodejs);
+
+  nodejs_defaults = modulemd_module_get_defaults(nodejs);
+  g_assert_nonnull(nodejs_defaults);
+  g_assert_null(modulemd_defaults_v1_get_default_stream(nodejs_defaults,NULL));
+  g_clear_object(&merged_index);
+
+  /* Get another set of objects that will override  g_assert_null(modulemd_defaults_v1_get_default_stream the above */
+  yaml_path = g_strdup_printf ("%s/overriding.yaml", g_getenv ("TEST_DATA_PATH"));
+  override_index = modulemd_module_index_new ();
+  modulemd_module_index_update_from_file (override_index, yaml_path, TRUE, &failures,&error);
+  g_clear_pointer(&yaml_path,g_free);
+
+  /*
+   * Test that override_index at a higher priority level succeeds
+   * Test that adding both of these at the same priority level fails
+   * with a merge conflict.
+   * Use randomly-selected high and low values to make sure we don't have
+   * sorting issues.
+   */
+  merger = modulemd_module_index_merger_new ();
+  random_low = rand()%100+1;
```

Comment:
Please look at this on how to use `random` with `GLib`. https://developer.gnome.org/glib/stable/glib-Random-Numbers.html

Declare your variable datatypes accordingly.